src/TreeSitter/Node.hs

```diff
@@ -33,7 +33,8 @@ data TSNode = TSNode !Word32 !Word32 !Word32 !Word32 !(Ptr ()) !(Ptr ())
 newtype Struct a = Struct { runStruct :: forall b . Ptr b -> IO (a, Ptr a) }
 
 evalStruct :: Struct a -> Ptr b -> IO a
-evalStruct = fmap (fmap fst) . runStruct
+evalStruct s p = fmap fst $! runStruct s p
+{-# INLINE evalStruct #-}
```

Comment:
Awesome 👍 For my own edification, I’d love to hear a breakdown (if you have one) of how much each of the different factors (inlining, avoiding partial application, and strict application) contribute, especially at `-O0`. Either way, great catch!